lib/logstash/outputs/bigquery/batcher.rb

```diff
@@ -49,15 +50,14 @@ def enqueue(message)
             if is_flush_request || length_met || size_met
               orig = @batch
               clear
-
-              yield(orig) if block_given?
-              return orig
             end
 
-            nil
           ensure
             @lock.write_lock.unlock
           end
+
+          yield(orig) if block_given? && !orig.nil?
```

Comment:
Seeing how (1) it copies the batch to `orig`, (2) doesn't mutate the object in `@batch`, (3) executes `clear` from within the lock, I think we're safe in `enqueue`.

As for [`publish`](https://github.com/logstash-plugins/logstash-output-google_bigquery/blob/7822aa688474148c1c5059f611879c86cda29411/lib/logstash/outputs/google_bigquery.rb#L232), we don't rely on any instance state, so as long as the google java client we create in [`initialize_google_client`](https://github.com/logstash-plugins/logstash-output-google_bigquery/blob/7822aa688474148c1c5059f611879c86cda29411/lib/logstash/outputs/bigquery/streamclient.rb#L95-L107) is thread safe, then I think we're good to merge here.